session/database.rst

```diff
@@ -0,0 +1,600 @@
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            # ...
+            Symfony\Component\HttpFoundation\Session\Storage\Handler\RedisSessionHandler:
+                arguments:
+                    - '@Redis'
+                    # you can optionally pass an array of options. The only option is 'prefix',
+                    # which defines the prefix to use for the keys to avoid collision on the Redis server
+                    # - { prefix: 'my_prefix' }
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <services>
+            <service id="Symfony\Component\HttpFoundation\Session\Storage\Handler\RedisSessionHandler">
+                <argument type="service" id="Redis"/>
+                <!-- you can optionally pass an array of options. The only option is 'prefix',
+                     which defines the prefix to use for the keys to avoid collision on the Redis server:
+                <argument type="collection">
+                    <argument key="prefix">my_prefix</argument>
+                </argument> -->
+            </service>
+        </services>
+
+    .. code-block:: php
+
+        // config/services.php
+        use Symfony\Component\HttpFoundation\Session\Storage\Handler\RedisSessionHandler;
+
+        $container
+            ->register(RedisSessionHandler::class)
+            ->addArgument(
+                new Reference('Redis'),
+                // you can optionally pass an array of options. The only option is 'prefix',
+                // which defines the prefix to use for the keys to avoid collision on the Redis server:
+                // ['prefix' => 'my_prefix'],
+            );
+
+Next, use the :ref:`handler_id <config-framework-session-handler-id>`
+configuration option to tell Symfony to use this service as the session handler:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/framework.yaml
+        framework:
+            # ...
+            session:
+                handler_id: Symfony\Component\HttpFoundation\Session\Storage\Handler\RedisSessionHandler
+
+    .. code-block:: xml
+
+        <!-- config/packages/framework.xml -->
+        <framework:config>
+            <!-- ... -->
+            <framework:session handler-id="Symfony\Component\HttpFoundation\Session\Storage\Handler\RedisSessionHandler"/>
+        </framework:config>
+
+    .. code-block:: php
+
+        // config/packages/framework.php
+        use Symfony\Component\HttpFoundation\Session\Storage\Handler\RedisSessionHandler;
+
+        // ...
+        $container->loadFromExtension('framework', [
+            // ...
+            'session' => [
+                'handler_id' => RedisSessionHandler::class,
+            ],
+        ]);
+
+That's all! Symfony will now use your Redis server to read and write the session data.
+
+Store Sessions in a Relational Database (MySQL, PostgreSQL)
```

Comment:
IMO this part should be before the Redis part
project mainly have a relational database, but should not have Redis for example

wdyt? is there a reason you write redis before?

Reply:
The order doesn't matter much because we display the table of contents at the beginning of the article. In any case, we also say this in the article introduction:

> Symfony can store sessions in all kinds of databases (relational, NoSQL and
key-value) but recommends key-value databases like Redis to get best performance.